mypy/subtypes.py

```diff
@@ -186,6 +190,24 @@ def visit_tuple_type(self, left: TupleType) -> bool:
         else:
             return False
 
+    def visit_typeddict_type(self, left: TypedDictType) -> bool:
+        right = self.right
+        if isinstance(right, Instance):
+            if right.type.typeddict_type is not None:
+                return is_subtype(left, right.type.typeddict_type, self.check_type_parameter)
+            return is_subtype(left.fallback, right, self.check_type_parameter)
+        elif isinstance(right, TypedDictType):
+            if not left.names_are_wider_than(right):
+                return False
+            for (_, l, r) in left.zip(right):
+                if not is_subtype(r, l, self.check_type_parameter):
```

Comment:
Item types should be invariant, since they are mutable. This would make them covariant -- use `is_equivalent` instead of `is_subtype`. Also add test case.